azure-proto-core/ArmClient.cs

```diff
@@ -202,11 +184,7 @@ public ResourceGroupOperations ResourceGroup(PhResourceGroup resourceGroup)
                 location = azure_proto_core.Location.Default;
             }
 
-            TContainer container;
-            if (!Registry.TryGetContainer<TContainer, ArmResource, TOperations, TResource>(this.ClientContext, new ArmResource($"/subscriptions/{subscription}/resourceGroups/{resourceGroup}", location), out container))
-            {
-                throw new InvalidOperationException($"No resource type matching '{typeof(TResource)}' found.");
-            }
+            TContainer container = Activator.CreateInstance(typeof(TContainer), ClientContext, new ArmResource($"/subscriptions/{subscription}/resourceGroups/{resourceGroup}", location)) as TContainer;
```

Comment:
This seems reasonable and better then having a manual registry. The only reason a manual registry might be interesting is if using reflection becomes a bottleneck but I don't think that will be the case for these. 

This is assuming that all the types you are creating has a particular constructor signature, do we have any guarantees around that? If not we might want to had some error handling here.

Reply:
We do have guarantees around that all of them will be auto-generated